Source/WebKit/UIProcess/API/wpe/WPEWebViewPlatform.h

```diff
@@ -81,14 +82,16 @@ class ViewPlatform final : public View {
     Vector<WebKit::WebPlatformTouchPoint> touchPointsForEvent(WPEEvent*);
 #endif
 
+    void dispatchPendingNextPresentationUpdateCallbacks();
+
     gboolean handleEvent(WPEEvent*);
     void handleGesture(WPEEvent*);
 
     GRefPtr<WPEView> m_wpeView;
     RefPtr<WebKit::AcceleratedBackingStoreDMABuf> m_backingStore;
     uint32_t m_displayID { 0 };
     unsigned long m_bufferRenderedID { 0 };
-    CompletionHandler<void()> m_nextPresentationUpdateCallback;
+    Vector<CompletionHandler<void()>> m_nextPresentationUpdateCallbacks;
```

Comment:
I find this Vector is used in a custom way: we insert at the front and retrieve from the back, the opposite of regular queues where you queue at the back and take from the front.

I understand we do this because only `takeLast` exists (and not `takeFirst`), but find that `m_next` is confusing: `at(0)` would actually be the `last` and not the `next`...

`m_frontInsertedPresentationUpdateCallbacks`?

Reply:
next is because this comes from `scheduleAfterNextPresentationUpdate` so all the callbacks are called the next presentation update.